R/spark_compile.R

```diff
@@ -45,11 +45,13 @@ spark_compile <- function(jar_name,
   jar_path <- file.path(java_path, jar_name)
 
   scala_path <- file.path(root, "java")
-  scala_files <- list.files(scala_path, pattern = "scala$", full.names = TRUE)
+  scala_files <- list.files(scala_path, pattern = "scala$",
+                            full.names = TRUE,
+                            recursive = TRUE)
 
   # apply user filter to scala files
   if (is.function(filter))
-    scala_files <- filter(scala_files)
+    scala_files <- Filter(filter, scala_files)
```

Comment:
I figured it's slightly easier to provide a function that handles one file, but given that this has been an exported function I probably shouldn't have messed with it. Will revert and modify `make_version_filter()`.